### ISSUE TITLE
Training inplementation

### ISSUE THREAD

**Issue by YixinZhu042** [zheng95z/rgbx#10]
Hi, thanks for your great work!
I am trying to follow your work on SD3.5-large, and I want to know how you design the training? 
I used a for-loop in one step to generate all 5 intrinsics, accumulate loss together and then back propagate. However, this design maybe results in huge memory use and causes a lot of serious and strange problems.  Even it cannot run on A100-80G.
At last, I choose 1 intrinsic randomly in one step and then training script can run. But I don't know whether it can generate accurate training results.

**Comment by zheng95z** [comment]
Hi @YixinZhu042,

We generate and train all intrinsics at one step. Also, the number of total intrinsics scales with batch size.

We did not encounter issues when finetuning SD 1.5 and SD 2.1.

I'm guessing the reason is that you're finetuning SD3.5-large, which is memory consuming.

Best,
